### ISSUE TITLE
gitlabCommitStatus missing required parameter

### ISSUE THREAD

**Issue by mcg** [jenkinsci/gitlab-plugin#795]
Might be related to #789
Seems like 1.5.8 is still broken for us around `gitlabCommitStatus`. We use it as described in the docs as:
```
 gitlabCommitStatus {
       // The result of steps within this block is what will be sent to GitLab 
       sh 'mvn install'
    }
```
And are getting:
```
Missing required parameter: "name" @ line 104, column 21.
                       gitlabCommitStatus{
```
Reverting back to 1.5.6 resolved the issue.

**Comment by omehegan** [comment]
@mcg thanks for letting us know, that does suggest that the simple form of the usage is broken. @stchar can you look at this?

**Comment by stchar** [comment]
so `name` is optional after all))) ok, I will take a look.  
As a workaround 
```
 gitlabCommitStatus('jenkins') { ... }
 gitlabCommitStatus(name: 'jenkins') { ... }
```

**Comment by mcg** [comment]
@stchar Thanks for looking. Unfortunately we use the simple form in many places. The best workaround for us is the downgrade to 1.5.6

**Comment by stchar** [comment]
@mcg 
Am I right that your pipelines look like
```
properties([
    gitLabConnection('test-connection')
])

node('master') {
  git 'http://gitlab/test/test.git'
  gitlabCommitStatus(){
    echo 'this is a single stage'
  }
}
```
So it is bare `gitlabCommitStatus` without `gitlabBuilds`

**Comment by mcg** [comment]
@stchar Correct. No `gitlabBuilds`.

**Comment by stchar** [comment]
@mcg, actually I'm asking as I've tested  the pipeline above in the docker containers described in this repository
gitlab 9.4.1
jenkins 2.89.2
GitLab_plugin 1.5.8

And it works

Moreover I can generate the snippet without `name` parameter in the syntax generator
![image](https://user-images.githubusercontent.com/1569458/42527963-c9688174-8482-11e8-8a68-ab6e177ba991.png)

**Comment by mcg** [comment]
Jenkins: 2.121.1
Gitlab: 10.8.2-ee
Works with gitlab_plugin 1.5.6, does not with 1.5.8.

Below is a snip that fails.
```
  pipeline {
        environment {
....
        }

        options {
            ansiColor('xterm')
            timestamps()
            timeout(time: 1, unit: 'HOURS')
            gitLabConnection('Gitlab')
        }

        agent { node { label 'core' } }

        stages {
            stage('Gather Components') {
                steps {
                    script {
                        gitLogEntry = sh(returnStdout: true, script: "git log -1")
                    }
                    gitlabCommitStatus{
...
                    }
                }
....
```

**Comment by stchar** [comment]
@mcg.  I've prepared the fix, you can find the plugin in PR CI job artifacts.
https://ci.jenkins.io/job/Plugins/job/gitlab-plugin/job/PR-799/1/artifact/target/gitlab-plugin-1.5.9-SNAPSHOT.hpi

Can you try it?

**Comment by mcg** [comment]
Will do, but can't until next week.

**Comment by mcg** [comment]
@stchar Looks like that fixes the issue for us.

**Comment by omehegan** [comment]
@mcg thanks for testing, I will merge this and get it released. @stchar thanks for the fix!

**Comment by omehegan** [comment]
This was released in version 1.5.9 of the plugin.
